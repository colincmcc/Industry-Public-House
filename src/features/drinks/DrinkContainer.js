import React from 'react'
import { Query } from 'react-apollo'
import gql from 'graphql-tag'
import { Switch, Route } from 'react-router-dom'
import { withRouter } from "react-router-dom";
import shortid from 'shortid'
import { withStyles } from '@material-ui/core/styles';

import Paper from '@material-ui/core/Paper';

import PageHeaderContainer from '../../common/components/page/PageHeaderContainer'
import DrinkNavComponent from './DrinkNavComponent'
import TapLIstComponent from '../../common/components/digitalPour/TapLIstComponent';
import bgImg from '../../common/assets/img/drinks_banner.jpg';
import FoodDrinkComponent from '../common/FoodDrinkComponent'
import DrinkMenuComponent from './DrinkMenuComponent'


const styles = theme => ({
  root: {
    ...theme.mixins.gutters(),
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2,
    margin: "2em",
    backgroundColor: "#F4EDDC",
    color: "#110C02"

  },
});
// TODO: get drinkNavItems and locations from Wordpress

const DrinkContainer = (props) => {
  const selectedDrinkType = props.selectedDrinkType
  const currentLocation = props.currentLocation


    // ! Location id's are set through Digital Pour. Not a good index reference, since they don't start with 0
    const locations = [
      {id: 1, label: "Lawrenceville", location: "lv"},
      {id: 2, label: "North Fayette", location: "nf"}
    ]
    const { classes } = props;

    return(
      <Query query={ DP_TAPS } variables={{location: currentLocation}}>
      {
        ({loading, error, data, client}) =>{

          const drinkNavItems = [
            {label: "Cocktails", slug: "Cocktails", component: <DrinkMenuComponent query={WP_COCKTAILS} drinkType="cocktails" />, showLocation: false},
            {label: "Taps", slug: "Taps", component: <TapLIstComponent taps={data.allTaps} loading={loading} error={error} />, showLocation: true, props: ""},
            {label: "Cans", slug: "Cans", component: <DrinkMenuComponent query={WP_CANS} drinkType="cans" />, showLocation: true},
            {label: "Wine", slug: "Wine", component: <DrinkMenuComponent query={WP_WINE} drinkType="wine" />, showLocation: false},
            {label: "Premium", slug: "Premium", component: <DrinkMenuComponent query={WP_PREMIUM} drinkType="premium" />, showLocation: true},
          ]

          return(

          <div>
            <PageHeaderContainer bgImg={bgImg} heading="Drinks" review={true} />

            <DrinkNavComponent client={client} locations={locations} navItems={drinkNavItems} currentLocation={currentLocation} selectedDrinkType={selectedDrinkType} />

            <Paper className={classes.root} elevation={2}>
            <Switch>
            <Route
            exact
            path="/Drink"
            render={() => <DrinkMenuComponent query={WP_COCKTAILS} drinkType="wine"/>}
          />
              {drinkNavItems.map((navItem) => (
                <Route key={shortid.generate()} exact path={'/Drink/' + navItem.slug} render={() => navItem.component} />
              ))}

            </Switch>
            </Paper>
          </div>
        )
        }
      }
  </Query>
    )
}

export default withStyles(styles)(withRouter(DrinkContainer))
const WP_COCKTAILS = gql`
  {
    menuItems: allCocktails {
      id
      acf {
        price
        name
        description
      }
    }
  }
`;

const WP_CANS = gql`
  {
    menuItems: allCans {
      id
      locations {
        id
      }
      acf {
        name
        price
        description
      }
    }
  }
`;

const WP_WINE = gql`
  {
    menuItems: allWine {
      id
      acf {
        name
        price
        description
      }
    }
  }
`;

const WP_PREMIUM = gql`
  {
    menuItems: allPremium {
      id
      locations {
        id
      }
      acf {
        tasting_notes
        proof
        aged
        background_picture
        distillery
        product_name
      }
    }
  }
`;

export const DP_TAPS = gql`
  query Taps($location: Int!) {
    allTaps(location: $location) {
      Id
      MenuItemProductDetail {
        BeverageType
        PosReportedPercentFull
        BeverageNameWithVintage
        Beverage {
          Brewery {
            BreweryName
            Location
          }
          Cidery {
            CideryName
            Location
          }
          Meadery {
            MeaderyName
            Location
          }
          CoffeeProducer {
            CoffeeProducerName
            Location
          }
          Abv
          Ibu
          HasAward
          StyleColor
          RateBeerUrl
          BeerAdvocateUrl
          UntappdUrl
          HopsUsed
          BarrelAging
          BeerStyle {
            StyleName
          }
        }
      }
    }
  }
`;