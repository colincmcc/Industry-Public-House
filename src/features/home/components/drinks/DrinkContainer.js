import React, { Component } from 'react'
import { Query } from 'react-apollo'
import gql from 'graphql-tag'


import DrinkNavComponent from './DrinkNavComponent'
import DrinkMenuContainer from './DrinkMenuContainer'

// TODO: get drinkNavItems and locations from Wordpress

const DrinkContainer = (props) => {
  const selectedDrinkType = props.cachedState.selectedDrinkType
  const currentLocation = props.cachedState.currentLocation


    const drinkNavItems = [
      {label: "Cocktails", slug: "cocktails", component: "CocktailContainer"},
      {label: "Taps", slug: "taps", component: "CocktailContainer"},
      {label: "Bottles", slug: "bottles", component: "CocktailContainer"},
      {label: "Wine", slug: "wine", component: "CocktailContainer"},
      {label: "Premium", slug: "premium", component: "CocktailContainer"},
    ]

    const locations = [
      {id: 1, label: "Lawrenceville", location: "lv"},
      {id: 2, label: "North Fayette", location: "nf"}
    ]
    return(
      <Query query={ DP_TAPS } variables={{location: currentLocation}}>
      {
        ({loading, error, data, client}) =>{
          if(loading) return <p>Loading...</p>
          if(error) return <p>Error</p>

          return(

          <div>
            <DrinkNavComponent client={client} locations={locations} navItems={drinkNavItems}  />
            <DrinkMenuContainer data={data} drinkMenu={selectedDrinkType} />
          </div>
        )
        }
      }
  </Query>
    )
}

export default DrinkContainer

export const DP_TAPS = gql`
  query Taps($location: Int!) {
    allTaps(location: $location) {
      Id
      MenuItemProductDetail {
        BeverageType
        PosReportedPercentFull
        BeverageNameWithVintage
        Beverage {
          Brewery {
            BreweryName
            Location
          }
          Cidery {
            CideryName
            Location
          }
          Meadery {
            MeaderyName
            Location
          }
          CoffeeProducer {
            CoffeeProducerName
            Location
          }
          Abv
          Ibu
          HasAward
          StyleColor
          RateBeerUrl
          BeerAdvocateUrl
          UntappdUrl
          HopsUsed
          BarrelAging
          BeerStyle {
            StyleName
          }
        }
      }
    }
  }
`;