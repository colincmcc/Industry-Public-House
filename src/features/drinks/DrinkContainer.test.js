import React from 'react';
import { shallow } from 'enzyme';
import DrinkContainer from './DrinkContainer';

describe('DrinkContainer', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<DrinkContainer debug />);

    expect(component).toMatchSnapshot();
  });
});
