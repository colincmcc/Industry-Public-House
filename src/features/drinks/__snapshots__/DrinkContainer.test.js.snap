// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrinkContainer should render correctly in "debug" mode 1`] = `
<Route>
  <Component />
</Route>
`;
